scripts/metavar.py

```diff
@@ -771,9 +784,21 @@ def intrinsic_elements(self, check_dict=None):
         Fortran does not support a way to reference those elements.
         """
         if self.is_ddt():
-            element_names = None
-            raise ValueError("shouldn't happen?")
-            # To Do, find and process named elements of DDT
+            dtitle = self.get_prop_value('type')
+            if ddt_lib and (dtitle in ddt_lib):
+                element_names = []
+                ddt_def = ddt_lib[dtitle]
+                for dvar in ddt_def.variable_list():
+                    delems = dvar.intrinsic_elements(check_dict=check_dict,
+                                                     ddt_lib=ddt_lib)
+                    if delems:
+                        element_names.extend(delems)
+                    # end if
+                # end for
+                if not element_names:
+                    element_names = None
+                # end if
+            # end if
```

Comment:
Maybe raise an `exception` above and below this line as else clauses? In the case where either else could be hit, I don't think `element_names` is defined.